### ISSUE TITLE
Idea: Predefined Wallet Top-Up Amounts

### ISSUE THREAD

**Issue by Buda9** [fmido88/moodle-enrol_wallet#18]
What do you think about this idea? Instead of users manually enter the amount they want to top up their wallet, we could have an option (or a separate page) where they can choose from predefined amounts, for example, buttons with packages of $10, $20, $40, and $80. Once they click an amount, a payment page would open for them where they could finish purchase. The practical application of this is for purchasing 1-on-1 package sessions with teachers at e.g. a language school. Example:

![Screenshot_120](https://github.com/fmido88/moodle-enrol_wallet/assets/1422662/2d36bb5d-1b9b-49ac-86c9-721ea48fa8f2)

or

![Screenshot_127](https://github.com/fmido88/moodle-enrol_wallet/assets/1422662/d7b6b938-7acf-4484-9204-1484dcb6fc76)

So basically, when someone clicks "package" and buy now button, it should open popup window to make a payment

Would it be to difficult to build this?

PS. Woo example of this is something like this https://woo.com/products/tiered-pricing-table-for-woocommerce/

**Comment by fmido88** [comment]
Hi @Buda9 
This is the first idea came to me but then I thought that it wouldn't be as practical as it sound due to the variation of courses prices and so on.
So, I created the conditional discount which will serve the same purpose but with more flexibility.
The only thing that is missing is to display to the user the offer (bundle like) the site offer by conditional discounts.
That is my next move.

**Comment by Buda9** [comment]
"So, I created the conditional discount which will serve the same purpose but with more flexibility."
Yes, this is such a great feature. Thank you for this.

"The only thing that is missing is to display to the user the offer (bundle like) the site offer by conditional discounts."
Great! I'm looking forward to it.

**Comment by fmido88** [comment]
Hello @Buda9 
I think that the new update is a good alternative to package

Also there is offers where the course creator can decide if the user is eligible to get this course with a discount or not

for example: Purchase course1 then you can get course2 for free

I think with offers and conditional discounts the course creator or admins could set any packages they like to users.

**Comment by Buda9** [comment]
Hi @fmido88 

Thank you. I updated the plugin, however, I'm having a bit difficulties to understand how exactly offers work. Should they be defined inside course enrolment page?

Offers page:
![Screenshot_216](https://github.com/fmido88/moodle-enrol_wallet/assets/1422662/2d70570d-89e6-46ea-8939-f1dd3fdbe49d)

Is this the correct place where we should define offers?
![Screenshot_217](https://github.com/fmido88/moodle-enrol_wallet/assets/1422662/8ab9fcfc-75cd-4063-9d5e-f6edccade9fa)

**Comment by fmido88** [comment]
If referral program is enabled, It will show the section of referral.
same for cashback.

The first section is for conditional discounts if there is any
![Screenshot 2024-02-17 173144](https://github.com/fmido88/moodle-enrol_wallet/assets/122031508/599ce29f-746f-4fd9-bc3a-1115d33b808a)

The section of free courses (has wallet enrollment with 0 cost)
![Screenshot 2024-02-17 173200](https://github.com/fmido88/moodle-enrol_wallet/assets/122031508/ea609f1f-d20c-44af-bd8d-64e3f5cabbd8)

And courses with available offers section
![Screenshot 2024-02-17 173221](https://github.com/fmido88/moodle-enrol_wallet/assets/122031508/22eaddd8-258b-4620-babb-8f44795e67ae)

**Comment by Buda9** [comment]
Me again. I saw on your Moodle repo page an interesting suggestion, quoting: "Can we make the wallet for a system using points/coins instead of currency with a calculation formula in the background? For instance, buy 1000 points would cost 15usd and buying 2000 points would cost 20usd. We also should display the wallet and use availability for points and not the currency? Thank you again! Keep up the great work."

I think implementing such a point system would be ideal for my case (language school). This approach would provide students with a simpler and more intuitive way of "paying" for classes and booking lessons, with the ability to better track their remaining points or credits for lessons.

However, from the developer perspective, I still don't know an idea how to implement such thing into Wallet plugin. Perhaps a separate plugin would be better for this?

**Comment by fmido88** [comment]
Hi @Buda9 
I did it (the first suggestion) 
When create or edit a conditional discount, you can add a bundle with a description with it, It will appear as a button so the user clicks it and charge his wallet with a faster way.
For the latest suggestion, I'll work on it but not now because it will take a lot of work ant time.

**Comment by Buda9** [comment]
Fantastic job! I can't thank you enough for this. It's even better than I expected it to be 🥇🥇🥇

This is how it looks on my site. I slightly modified order of the span's in template file. You'll notice it looks very similar to examples above

![Screenshot_225](https://github.com/fmido88/moodle-enrol_wallet/assets/1422662/09d65992-a9ac-4535-bf50-433ca8ae24a3)

**Comment by fmido88** [comment]
Hello @Buda9
I'm afraid if I did so, the amount is something like 42.05 and in some currencies this 5 cents is significant.
What I tried to do but I really didn't try it hard is to make cents in a smaller font size.
By the way your modification looks so elegant, great work.
